src/chapters/exceptions/exceptions-control-flow.rst

```diff
@@ -1,71 +1,137 @@
+is an exception. We can tell JavaScript to *try* to run a block of code and if an exception is thrown, to *catch* the
+exception and run a different block of code. Anticipating and catching the exception makes the exception now part of
+the control flow.
+
+.. note:: Catching an exception is also known as *handling* an exception.
 
 .. admonition:: Example
 
-   .. sourcecode:: js
+   In this example there is an array of animals. The user is asked to enter the index for the animal they want to see.
+   If the user enters an index that does NOT contain an animal, the code will try throw an ``TypeError`` when ``name``
+   is referenced on an undefined value.
+
+   There is a ``try`` block around the code that will throw the ``TypeError``. There is a ``catch`` block that catches the error
+   and contains code to inform the user that they entered an invalid index.
+
+   .. replit:: js
+      :slug: control-flow-type-error
       :linenos:
 
+      const input = require('readline-sync');
+
+      let animals = [{name: 'cat'}, {name: 'dog'}];
+      let index = Number(input.question("Enter index of animal:"));
+
       try {
-          console.log(x[0]);
-      }
-      catch(ReferenceError) {
-          console.log("We caught a ReferenceError, but our program continues to run!");
-          console.log("0");
+         console.log('animal at index:', animals[index].name);
+      } catch(TypeError) {
+         console.log("We caught a TypeError, but our program continues to run!");
+         console.log("You tried to access an animal at index:", index);
       }
 
+      console.log("the code goes on...");
+
    **Console Output**
 
+   If the user enters ``9``:
+   ::
+
+      Enter index of animal: 9
+      We caught a TypeError, but our program continues to run!
+      You tried to access an animal at index: 9
+      the code goes on...
+
+   If the user enters ``0``:
    ::
 
-      We caught a ReferenceError, but our program continues to run!
-      0
+      Enter index of animal: 0
+      animal at index: cat
+      the code goes on...
 
-   In this example we attempt to print out the first element of our variable ``x``. We forgot to define x, but nested our code inside of a ``try`` statement and specifically **catch** ``ReferenceErrors``. It should be noted that only ReferenceErrors will be caught by the catch statement. However, we can define as many catch statements as we want.
+.. tip:: ``catch`` blocks only execute if an exception is thrown
 
-JavaScript also provides us with a ``finally`` statement, a block of code that always runs every time regardless if the ``try``, or ``catch`` blocks run.
+
+Finally
+-------
+.. index:: ! finally
+
+.. index::
+   single: try catch; finally
+
+JavaScript also provides a ``finally`` block which must be used with ``try`` and ``catch`` blocks.  A ``finally`` block code runs after the ``try`` and ``catch``.
```

Comment:
Does ``finally`` HAVE to be used with ``try/catch``?  If not, then the word "must" should be replaced with "can".